tests/lib/cli.js

```diff
@@ -137,6 +137,43 @@ describe("cli", () => {
 
         });
 
+        describe("flat config", () => {
+            const originalCwd = process.cwd;
+            const originalEnv = { ...process.env };
+
+            afterEach(() => {
+                process.cwd = originalCwd;
+                process.env = { ...originalEnv };
+            });
+
+            it(`should use it when an eslint.config.js is present and useFlatConfig is true:${configType}`, async () => {
+                process.cwd = getFixturePath;
+
+                const exitCode = await cli.execute(`--no-ignore --ext .js ${getFixturePath("files")}`, null, useFlatConfig);
+
+                // When flat config is used, we get an exit code of 2 because the --ext option is unrecognized.
+                assert.strictEqual(exitCode, useFlatConfig ? 2 : 0);
+            });
+
+            it(`should not use it when ESLINT_USE_FLAT_CONFIG=false even if an eslint.config.js is present:${configType}`, async () => {
+                process.cwd = getFixturePath;
+                process.env.ESLINT_USE_FLAT_CONFIG = "false";
+
+                const exitCode = await cli.execute(`--no-ignore --ext .js ${getFixturePath("files")}`, null, useFlatConfig);
+
+                assert.strictEqual(exitCode, 0);
+            });
+
+            it(`should use it when ESLINT_USE_FLAT_CONFIG=true and useFlatConfig is true even if an eslint.config.js is not present:${configType}`, async () => {
+                process.env.ESLINT_USE_FLAT_CONFIG = "true";
+
+                const exitCode = await cli.execute(`--no-ignore --ext .js ${getFixturePath("files")}`, null, useFlatConfig);
+
+                // When flat config is used, we get an exit code of 2 because the --ext option is unrecognized.
+                assert.strictEqual(exitCode, useFlatConfig ? 2 : 0);
+            });
+        });
```

Comment:
Are we sure that in this setup an `eslint.config.js` file is not present? If this is executed in the project root, then there will be  https://github.com/eslint/eslint/blob/main/eslint.config.js

Reply:
Yup an `eslint.config.js` will not be present due to [this logic](https://github.com/eslint/eslint/blob/b0ee848602ed86dfad815fb819da420c7842552e/tests/lib/cli.js#L86-L99).

Reply:
I tried to run this test with the `process.env.ESLINT_USE_FLAT_CONFIG = "true";` line removed, and it doesn't fail. Only when I also remove [eslint.config.js](https://github.com/eslint/eslint/blob/main/eslint.config.js) file the test fails, which indicates that the file is present in the directory where the test is run.

Reply:
Good catch! There were two issues in the test:
1. I didn't set `process.cwd = getFixturePath` so it was still searching inside the git repository and finding [eslint.config.js](https://github.com/eslint/eslint/blob/main/eslint.config.js)
2. Once I set `process.cwd = getFixturePath`, it still was finding it because there's an `eslint.config.js` in `fixtures` as well. So I went ahead and wrote `process.cwd = () => getFixturePath("..")`

I've confirmed that the test now fails without `process.env.ESLINT_USE_FLAT_CONFIG = "true";`.